lib/remote_system.rb

```diff
@@ -150,8 +150,12 @@ def check_retrieve_files_dependencies
   end
 
   # Reads a file from the System. Returns nil if it does not exist.
-  def read_file(file)
-    run_command("cat", file, stdout: :capture, privileged: true)
+  def read_file(file, options = {})
+    if options[:privileged]
+      run_command("cat", file, stdout: :capture, privileged: true)
```

Comment:
How about

```ruby
def read_file(file, options = {})
  command_options = {
    stdout: :capture,
    privileged: false
  }.merge(options)

  run_command("cat", file, command_options)
...
```

That makes the intention to run unprivileged by default explicit and removes a conditional and some duplication.